core/src/main/java/org/apache/iceberg/avro/AvroSchemaUtil.java

```diff
@@ -90,6 +90,23 @@ static boolean hasIds(Schema schema) {
     return AvroCustomOrderSchemaVisitor.visit(schema, new HasIds());
   }
 
+  /**
+   * @param schema an Avro Schema
+   * @return true/false based on whether any of the nodes in the provided schema is missing an
+   * ID property recognizable by Iceberg core API. To have an ID recognizable by Iceberg core API:
+   * <ul>
+   *   <li>a field node under struct (record) schema should have {@link FIELD_ID_PROP} property
+   *   <li>an element node under list (array) schema should have {@link ELEMENT_ID_PROP} property
+   *   <li>a pair of key and value node under map schema should have {@link KEY_ID_PROP} and
+   *   {@link VALUE_ID_PROP} respectively
+   *   <li>a primitive node is not assigned any ID related properties
+   * </ul>
+   * @implNote see {@link MissingIds} for more details
```

Comment:
This is an internal class, so it shouldn't be referenced in Javadoc. Anything that is relevant should be noted here since this is the public facing part.

Reply:
Make sense, removed.